src/mapper/pkg/dnscache/dns_cache.go

```diff
@@ -38,6 +39,17 @@ func (d *DNSCache) GetResolvedIPs(dnsName string) []string {
 	return entry
 }
 
+func (d *DNSCache) GetMatchingEntriesForWildcard(dnsName string) []string {
```

Comment:
suggestion: change to `GetMatchingIPsForWildcard`, since that's what you actually need on the caller side.